### ISSUE TITLE
various errors when testing on windows + python3.5.2 (Anaconda 4.1.1 32bit)

### ISSUE THREAD

**Issue by lmiguelmh** [OpenIDC/pyoidc#267]
After following the documentation for the installation procedure (http://pyoidc.readthedocs.io/en/latest/install.html) I ran `py.test`:

```
________________________________ test_stateID _________________________________

    def test_stateID():
        seed = rndstr().encode("utf-8")
        sid0 = stateID("http://example.com/home", seed)
        sid1 = stateID("http://example.com/home", seed)
        assert sid0
        assert sid1
>       assert sid0 != sid1
E       assert 'd0dc36847cb92c3fc11fb036f852b35f' != 'd0dc36847cb92c3fc11fb036f852b35f'

test_oauth2_consumer.py:89: AssertionError
_______________________ TestOICConsumer.test_begin_file _______________________

self = <test_oic_consumer.TestOICConsumer object at 0x05961F70>
tmpdir = local('C:\\Users\\...\\AppData\\Local\\Temp\\...\\pytest-5\\test_begin_file0')

    def test_begin_file(self, tmpdir):
        path = tmpdir.strpath
        self.consumer.consumer_config["request_method"] = "file"
        self.consumer.consumer_config["temp_dir"] = path
        self.consumer.consumer_config["temp_path"] = path
        self.consumer.consumer_config["authz_page"] = "/authz"
        srv = Server()
        srv.keyjar = SRVKEYS

        sid, location = self.consumer.begin("openid", "code",
                                            path="http://localhost:8087")

        with responses.RequestsMock() as rsps:
            p = urlparse(self.consumer.request_uri)
            rsps.add(rsps.GET, self.consumer.request_uri,
>                    body=open(p.path).read(), status=200,
                     content_type='application/urlencoded')
E           FileNotFoundError: [Errno 2] No such file or directory: '/11MqCzIdwV'

test_oic_consumer.py:187: FileNotFoundError
____________________ TestProvider.test_provider_key_setup _____________________

self = <test_oic_provider.TestProvider object at 0x05999570>
tmpdir = local('C:\\Users\\...\\AppData\\Local\\Temp\\...\\pytest-5\\test_provider_key_setup0')

    def test_provider_key_setup(self, tmpdir):
        path = tmpdir.strpath
        provider = Provider("pyoicserv", SessionDB(SERVER_INFO["issuer"]), None,
                            None, None, None, None, "")
        provider.baseurl = "http://www.example.com"
>       provider.key_setup(path, path, sig={"format": "jwk", "alg": "RSA"})

test_oic_provider.py:756:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Anaconda3\lib\site-packages\oic-0.9.4.0-py3.5.egg\oic\oic\provider.py:1973: in key_setup
    fqdn=self.hostname, sig=sig, enc=enc)
C:\Anaconda3\lib\site-packages\oic-0.9.4.0-py3.5.egg\oic\utils\keyio.py:840: in key_export
    os.makedirs(local_path)
C:\Anaconda3\lib\os.py:231: in makedirs
    makedirs(head, mode, exist_ok)
...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = './C:', mode = 511, exist_ok = False

    def makedirs(name, mode=0o777, exist_ok=False):
        """makedirs(name [, mode=0o777][, exist_ok=False])

        Super-mkdir; create a leaf directory and all intermediate ones.  Works like
        mkdir, except that any intermediate path segment (not just the rightmost)
        will be created if it does not exist. If the target directory already
        exists, raise an OSError if exist_ok is False. Otherwise no exception is
        raised.  This is recursive.

        """
        head, tail = path.split(name)
        if not tail:
            head, tail = path.split(head)
        if head and tail and not path.exists(head):
            try:
                makedirs(head, mode, exist_ok)
            except FileExistsError:
                # Defeats race condition when another thread created the path
                pass
            cdir = curdir
            if isinstance(tail, bytes):
                cdir = bytes(curdir, 'ASCII')
            if tail == cdir:           # xxx/newdir/. exists if xxx/newdir exists
                return
        try:
>           mkdir(name, mode)
E           OSError: [WinError 123] El nombre de archivo, el nombre de directorio o la sintaxis de la etiqueta del volumen no son correctos: './C:'

C:\Anaconda3\lib\os.py:241: OSError
____________________________ test_time_a_while_ago ____________________________

    def test_time_a_while_ago():
        dt = datetime.utcnow()
        t = time_a_while_ago(seconds=10)
        delta = dt - t  # slightly less than 10
>       assert delta.seconds == 9
E       assert 10 == 9
E        +  where 10 = datetime.timedelta(0, 10).seconds
```

Is something wrong with my installation? Or I need to change something... 
Thanks,

**Comment by lwm** [comment]
I would guess that the current PyPi release is totally buggy. For example, the final failure was fixed in https://github.com/rohe/pyoidc/commit/4733696da12f6dd638c903452d4c8cb8fdec27ee.  I've asked for a new release in https://github.com/rohe/pyoidc/issues/264. Please chime in there to push the maintainer.

To avoid this, I install from sources with pip. It looks like:

```
$ pip install -e git://github.com/rohe/pyoidc.git#egg=oic
```

**Comment by lwm** [comment]
@roh, this is another reason to make a PyPi release. 

https://pypi.python.org/pypi/oic/0.9.4.0 is borked.

**Comment by schlenk** [comment]
Some of those are fixed in #353

**Comment by tpazderka** [comment]
Most of this has already been fixed. The only remainin known bug on Windows is split to #352 so I am closing this one.
